deps/openssl/config/opensslconf.h

```diff
@@ -1,324 +1,121 @@
+    | mipsel     | __MIPSEL__        |
+    | x32        | __ILP32__         |
+    | x64        | __x86_64__        |
+    | x64(win)   | _M_X64            |
+
+    These are the list which is not implemented yet.
+
+    | --dest-os | --dest-cpu | OpenSSL target arch  | CI  |
+    | --------- | ---------- | -------------------- | --- |
+    | linux     | mips       | linux-mips32,linux-mips64,linux64-mips64? | --- |
+    | linux     | mipsel     | ?                    | --- |
+    | android   | ia32       | android-x86          | --- |
+    | android   | arm        | android-armv7        | --- |
+    | android   | mips       | android-mips         | --- |
+    | android   | mipsel     | ?                    | --- |
+
+    Supported target arch list in OpenSSL can be obtained by typing
+    `deps/openssl/openssl/Configure LIST`.
+*/
+
+#undef OPENSSL_LINUX
+#if defined(__linux) && !defined(__ANDROID__)
+  #define OPENSSL_LINUX 1
 #endif
 
-#if defined(HEADER_NEW_DES_H) || defined(HEADER_DES_H)
- /* If this is set to 'unsigned int' on a DEC Alpha, this gives about a
-  * %20 speed up (longs are 8 bytes, int's are 4). */
-# undef DES_LONG
-# define DES_LONG unsigned int
-#endif
-
-#if defined(HEADER_BN_H) && !defined(CONFIG_HEADER_BN_H)
-# define CONFIG_HEADER_BN_H
-
- /* Should we define BN_DIV2W here? */
-
- /* Only one for the following should be defined */
- /* The prime number generation stuff may not work when
-  * EIGHT_BIT but I don't care since I've only used this mode
-  * for debuging the bignum libraries */
-# undef SIXTY_FOUR_BIT_LONG
-# undef SIXTY_FOUR_BIT
-# undef THIRTY_TWO_BIT
-# undef SIXTEEN_BIT
-# undef EIGHT_BIT
-# if defined(_M_X64) || defined(__aarch64__) || defined(__x86_64__)
-#  if defined(_LP64)
-#   define SIXTY_FOUR_BIT_LONG
-#  else
-#   define SIXTY_FOUR_BIT
-#  endif
-# elif defined(_M_IX86) || defined(__i386__) || defined(__arm__) || defined(__mips__)
-#  define THIRTY_TWO_BIT
-# endif
+#if defined(OPENSSL_LINUX) && defined(__i386__)
+  #include "./archs/linux-elf/opensslconf.h"
+#elif defined(OPENSSL_LINUX) && defined(__ILP32__)
+  #include "./archs/linux-x32/opensslconf.h"
+#elif defined(OPENSSL_LINUX) && defined(__x86_64__)
+  #include "./archs/linux-x86_64/opensslconf.h"
+#elif defined(OPENSSL_LINUX) && defined(__arm__)
+  #include "./archs/linux-armv4/opensslconf.h"
+#elif defined(OPENSSL_LINUX) && defined(__aarch64__)
+  /* Not Supported Yet */
+#elif defined(__APPLE__) && defined(__MACH__) && defined(__i386__)
+  #include "./archs/darwin-i386-cc/opensslconf.h"
+#elif defined(__APPLE__) && defined(__MACH__) && defined(__x86_64__)
+  #include "./archs/darwin64-x86_64-cc/opensslconf.h"
+#elif defined(_WIN32) && defined(_M_IX86)
+  #include "./archs/VC-WIN32/opensslconf.h"
+#elif defined(_WIN32) && defined(_M_X64)
+  #include "./archs/VC-WIN64A/opensslconf.h"
+#elif (defined(__FreeBSD__) || defined(__OpenBSD__)) && defined(__i386__)
+  #include "./archs/BSD-x86/opensslconf.h"
```

Comment:
To be clear, the uppercase directory name is generated by openssl's Configure script?

Style nit: can you write this as `#include` or `# include`?  This indent style isn't used anywhere else.